### ISSUE TITLE
Does not produce correct times for dates

### ISSUE THREAD

**Issue by mrdrbeaver** [jakubroztocil/rrule#46]
**Given:**
DTSTART: 20070311T020000 
RRULE: FREQ=YEARLY;BYMONTH=3;BYDAY=2SU

**Produces:** 
Sun Mar 11 2007 18:00:00 GMT-0700 (PDT),
Sun Mar 09 2008 18:00:00 GMT-0700 (PDT),
Sun Mar 08 2009 18:00:00 GMT-0700 (PDT),
Sun Mar 14 2010 18:00:00 GMT-0700 (PDT),
...
**Please Note:**
Times are incorrect with respect to the DTSTART

**Code to reproduce:**

``` node
var rrule = require('rrule').RRule;
var options = rrule.parseString('FREQ=YEARLY;BYMONTH=3;BYDAY=2SU');

 if (options) {
    options.dtstart = new Date('2007-03-11T02:00:00');
    var RRule = new rrule(options);
    console.log(RRule.all());
    done();
}
```

**Comment by etrubenok** [comment]
Try to run these tests in an OS which has a system timezone in UTC.

Otherwise, the JavaScript Date.getDate() returns an incorrect date of a month in function _iter: function(iterResult) { ... day = dtstart.getDate(), ... }. It actually returns a date in your system timezone, not in UTC as you would expect.

This was a root cause in my case when I got Fridays instead of Saturdays for the following RRULE:
'FREQ=WEEKLY;UNTIL=20150214T235959Z;INTERVAL=1;BYDAY=SA;WKST=MO;DTSTART=20150117T220000Z';

It works fine on the server with the UTC system timezone.

**Comment by davidgoli** [comment]
This was probably fixed with #235. Please reopen if you are still experiencing issues.
